Application/Genre/genre.rb

```diff
@@ -1,15 +1,18 @@
+require_relative '../item/item'
+
 class Genre
-  attr_accessor :name
-  attr_reader :id, :items
+  attr_accessor :id, :name
+  attr_reader :items
 
-  def initialize(_id, name)
-    @id = Random.rand(2..2000)
+  def initialize(id, name)
+    @id = id
     @name = name
     @items = []
   end
 
+  # Custom setter method to add an item
   def add_item(item)
-    item.genre = self
     @items << item
+    item.genre = self
```

Comment:
It would be great if you could change the code back here to add the genre property to the item first before the item gets added to the items array. That will make sure that all the item elements in the items array will have the genre properties updated value as self.